website/docs/Installation.md

```diff
@@ -31,19 +31,19 @@ The following command will deactivate the current `conda` environment:
 conda deactivate
 ```
 
-### Option 3: poetry
+### Option 3: pdm
 
-Another option is with `poetry`, which is a dependency manager for Python.
+Another option is with `pdm`, which is a dependency manager for Python.
 
-[Poetry](https://python-poetry.org/docs/) is a tool for dependency management and packaging in Python. It allows you to declare the libraries your project depends on and it will manage (install/update) them for you. Poetry offers a lockfile to ensure repeatable installs, and can build your project for distribution.
+[PDM](https://python-poetry.org/docs/) is a tool for dependency management and packaging in Python. It allows you to declare the libraries your project depends on and it will manage (install/update) them for you. PDM offers a lockfile to ensure repeatable installs, and can build your project for distribution.
 
-You can install it by following [this doc](https://python-poetry.org/docs/#installation),
+You can install it by following [this doc](https://pdm-project.org/latest/#installation),
 and then create a virtual environment as below:
 ```bash
-poetry init
-poetry shell
+pdm init
```

Comment:
Is this needed for an existing project?

Reply:
If it's an existing pdm project, it's not needed. I assume if the user has an existing pdm project, they will know this. It won't hurt their pyproject.toml even if they run it on an existing pdm project though.

When you run `pdm init` it first looks for things like pyproject.toml, setup.py, and requirements.txt. If a pyproject.toml already exists, it does not delete anything from it. Instead, if it's for instance a poetry project, it will convert the pyproject.toml from poetry to PEP 621. Note: It will not delete the user's existing poetry directives. Example:

```toml
[tool.poetry]
name = "example-project"
version = "0.1.0"
description = "An example Python project."
authors = ["Your Name <you@example.com>"]
license = "MIT"

# Specify the Python versions compatible with your project
[tool.poetry.dependencies]
python = "^3.8"
requests = "^2.25.1"
numpy = "^1.19.4"

# Development dependencies can be specified here
[tool.poetry.dev-dependencies]
pytest = "^6.1.2"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

becomes:

```toml
[tool.poetry]
name = "example-project"
version = "0.1.0"
description = "An example Python project."
authors = ["Your Name <you@example.com>"]
license = "MIT"

# Specify the Python versions compatible with your project
[tool.poetry.dependencies]
python = ">=3.11,<3.12"
requests = "^2.25.1"
numpy = "^1.19.4"

# Development dependencies can be specified here
[tool.poetry.dev-dependencies]
pytest = "^6.1.2"


[tool.pdm]
package-type = "application"

[tool.pdm.dev-dependencies]
dev = [
    "pytest<7.0.0,>=6.1.2",
]

[tool.pdm.build]
includes = []
[project]
name = "example-project"
version = "0.1.0"
description = "An example Python project."
authors = [
    {name = "bitnom", email = "14287229+bitnom@users.noreply.github.com"},
    {name = "Your Name", email = "you@example.com"},
]
dependencies = [
    "requests<3.0.0,>=2.25.1",
    "numpy<2.0.0,>=1.19.4",
]
requires-python = ">=3.11,<3.12"
readme = "README.md"
license = {text = "MIT"}
```

If it's a pip or any other project with setup.py and/or requirements.txt, it prompts the user, offering to import the deps to pyproject.toml.

If you try to `pdm add pyautogen` in a non-pdm project, you will get:

```
[ProjectError]: The pyproject.toml has not been initialized yet. You can do this by running `pdm init`.
```

Reply:
I might be wrong, but this section is not about setting up a project that uses autogen but about installing autogen. I feel that instructions for setting up a pdm-based project are outside of the scope of this document.